src/nvim/testdir/test_windows_home.vim

```diff
@@ -60,18 +60,18 @@ func Test_WindowsHome()
     let $USERPROFILE = 'unused'
     let $HOMEDRIVE = 'C:'
     let $HOMEPATH = '\foobar'
-    let $HOME = ''  " Force recomputing "homedir"
+    unlet $HOME  " Force recomputing "homedir"
```

Comment:
If plugins/tests really expect the Vim behavior perhaps we should mimic it after all. @blueyed

Reply:
Yes, plugins are an issue - it is bad if they see different behavior in subtle areas.
But for this test: this could/should be changed like this in Vim itself already - I assume it is from before `unlet` supported envs.

Reply:
@erw7 
Why is this change necessary?

Reply:
This is because vim deletes environment variables with `let $FOO=''` on Windows. It seems that there is no way to create an empty environment variable other than calling the API on Windows. Therefore, I changed it because it seems that the same operation as vim is better.

Reply:
I see.  We need `unlet` to unset it, because setting it to an empty value does not unset it, right?
See also the discussion in https://github.com/pdbpp/pdbpp/pull/268 (for `exists("$EMPTY")`).

Reply:
Given your latest commit this could use the Vim version of the test already, couldn't it?

Reply:
Yes, no test changes are needed.